### PR TITLE
Feature update remove

### PR DESCRIPTION
**Pull request by vadiminc**

### CODE REVIEW

File: contracts/ISSVNetwork.sol [hunk 1]
```diff
@@ -72,6 +72,12 @@ interface ISSVNetwork {
      */
     event OperatorAdded(string name, address ownerAddress, bytes publicKey);
 
+    /**
+     * @dev Emitted when the operator has been deleted.
+     * @param publicKey Operator's Public Key.
+     */
+    event OperatorDeleted(bytes publicKey);
```
[the remark this answers is not in the document]
Comment (vadiminc) [reply contracts/ISSVNetwork.sol]: done

File: contracts/ISSVNetwork.sol [hunk 2]
```diff
@@ -84,6 +90,24 @@ interface ISSVNetwork {
...
+        address ownerAddress,
+        bytes publicKey,
+        Oess[] oessList
+    );
+
+    /**
+     * @dev Emitted when the validator has been deleted.
+     * @param publicKey Operator's Public Key.
+     */
+    event ValidatorDeleted(bytes publicKey);
```
[the remark this answers is not in the document]
Comment (vadiminc) [reply contracts/ISSVNetwork.sol]: done

File: contracts/ISSVNetwork.sol [hunk 3]
```diff
@@ -98,4 +122,34 @@ interface ISSVNetwork {
...
+     */
+    function updateValidator(
+        bytes calldata _publicKey,
+        bytes[] calldata _operatorPublicKeys,
+        bytes[] calldata _sharesPublicKeys,
+        bytes[] calldata _encryptedKeys
+    ) external;
+
+    /**
+     * @dev [Ownable] Deletes a validator from the list.
```
[the remark this answers is not in the document]
Comment (vadiminc) [reply contracts/ISSVNetwork.sol]: removed, it was just comment

File: contracts/ISSVNetwork.sol [hunk 4]
```diff
@@ -98,4 +122,34 @@ interface ISSVNetwork {
...
+     * @param _publicKey Validator public key.
+     */
+    function deleteValidator(
+        bytes calldata _publicKey
+    ) external;
+
+    /**
+     * @dev [Ownable] Deletes an operator from the list.
```
[the remark this answers is not in the document]
Comment (vadiminc) [reply contracts/ISSVNetwork.sol]: removed, it was just comment

File: contracts/SSVNetwork.sol [hunk 5]
```diff
@@ -71,4 +81,67 @@ contract SSVNetwork is ISSVNetwork {
...
+    /**
+     * @dev See {ISSVNetwork-updateValidator}.
+     */
+    function updateValidator(
+        bytes calldata _publicKey,
+        bytes[] calldata _operatorPublicKeys,
+        bytes[] calldata _sharesPublicKeys,
+        bytes[] calldata _encryptedKeys
+    ) onlyValidator(_publicKey) public virtual override {
+        require(_publicKey.length == 48, "Invalid public key length");
```
[the remark this answers is not in the document]
Comment (vadiminc) [reply contracts/SSVNetwork.sol]: done

File: contracts/SSVNetwork.sol [hunk 6]
```diff
@@ -71,4 +81,67 @@ contract SSVNetwork is ISSVNetwork {
...
+    }
+
+    /**
+     * @dev See {ISSVNetwork-deleteValidator}.
+     */
+    function deleteValidator(
+        bytes calldata _publicKey
+    ) onlyValidator(_publicKey) public virtual override {
+        require(
+            validators[_publicKey].ownerAddress != address(0),
```
[the remark this answers is not in the document]
Comment (vadiminc) [reply contracts/SSVNetwork.sol]: done

File: contracts/SSVNetwork.sol [hunk 7]
```diff
@@ -71,4 +81,67 @@ contract SSVNetwork is ISSVNetwork {
...
+    }
+
+    /**
+     * @dev See {ISSVNetwork-deleteOperator}.
+     */
+    function deleteOperator(
+        bytes calldata _publicKey
+    ) onlyOperator(_publicKey) public virtual override {
+        require(
+            operators[_publicKey].ownerAddress != address(0),
```
[the remark this answers is not in the document]
Comment (vadiminc) [reply contracts/SSVNetwork.sol]: done

**Review by adamzigdon — APPROVED** [review]

**Pull request merged**